File: src/Weekly_Contest_181/Longest_Happy_Prefix/Solution.py


# https://leetcode.com/contest/weekly-contest-181/problems/longest-happy-prefix/
# For some reason this brute force solution works
class Solution:
    def longestPrefix(self, s: str) -> str:
        l = len(s)
        final = ''
        for i in range(l):
            if s[:i] == s[l - i:]:
                final = s[:i]
        return final

File: src/Weekly_Contest_181/Longest_Happy_Prefix/Solution.kt
package Weekly_Contest_181.Longest_Happy_Prefix

fun main() {
    println(Solution().longestPrefix("a"))
}

//https://leetcode.com/contest/weekly-contest-181/problems/longest-happy-prefix/
//timeout
class Solution {
    fun sameContent(a: ByteArray, b: ByteArray, l: Int): Boolean {
        val startIndex = b.size - l
        for (i in 0 until l) {
            if (a[i] != b[startIndex + i]) return false
        }
        return true
    }

    fun longestPrefix(s: String): String {
        val array = s.toByteArray()

        for (l in s.length - 1 downTo 0) {
            if (sameContent(array, array, l)) {
                return s.substring(0, l)
            }
        }
        error("wtf")
    }
}